lib/clientBuilder.js

```diff
@@ -79,8 +79,21 @@ function OktaAuthBuilder(args) {
 
   sdk.token = {
     getWithoutPrompt: util.bind(token.getWithoutPrompt, sdk, sdk),
+    getWithPopup: util.bind(token.getWithPopup, sdk, sdk),
+    getWithRedirect: util.bind(token.getWithRedirect, sdk, sdk),
+    parseFromUrl: util.bind(token.parseFromUrl, sdk, sdk),
     decode: util.bind(token.decodeToken, sdk)
   };
+
+  // This is exposed so we can set window.location in our tests
+  sdk.token.getWithRedirect._setLocation = function(url) {
```

Comment:
Another option is to create some util helper functions "util.redirect" and "util.getLocation" - in that case, we wouldn't be able to easily test those methods, but it would be easy to test everything that depends on them, i.e. would be able to mock without having to expose private methods.